### PR TITLE
Pixelcnn

### PR DESCRIPTION
**Pull request by timonpalm**
Die Conditioned Gated PixelCNN, die am Besten funktionierte.

Die Implementierung habe ich auf MNIST getestet, um die Generierung auch iwie beurteilen zu können. Habe leider noch nicht die wandb integration eingebaut, daher nur die Ergebnisse bei mir.

features:
- reconstruction
- class conditioned sampling

Das PixelCNN hat ein leicht abgewandeltes training, daher hab ich mir erstaml einen train funktion neu geschrieben. Wäre super das noch zu vereinheitlichen mit den anderen Models. Key difference ist glaube, dass man jetzt noch die class indices braucht

### CODE REVIEW

File: src/pml_vqvae/vqvae/pixel_cnn.py [hunk 1]
```diff
@@ -1,37 +1,380 @@
...
     @staticmethod
-    def visualize_output(batch, output, target, prefix: str = "", base_dir: str = "."):
+    def visualize_output(batch, output, prefix: str = "", base_dir: str = "."):
         show(batch, outfile=os.path.join(base_dir, f"{prefix}_original.png"))
         show(
             output,
             outfile=os.path.join(base_dir, f"{prefix}_reconstruction.png"),
         )
 
+    def training(
```
Comment (rosinusserrano) [inline src/pml_vqvae/vqvae/pixel_cnn.py]: ich würde die funktion umbennen, nn.Module.training ist eigentlich ein attribut das sagt ob das model in train oder eval mode ist.

Comment (timonpalm) [reply src/pml_vqvae/vqvae/pixel_cnn.py]: Nach meinem Wissen ist das die Funktion train() mit counterpart eval(). Hab gerade nochmal geguckt und konnte auch keine training finden

Comment (rosinusserrano) [reply src/pml_vqvae/vqvae/pixel_cnn.py]: https://pytorch.org/docs/stable/generated/torch.nn.Module.html

Keine funktion sondern ein attribut
